src/spesh/optimize.c

```diff
@@ -2772,10 +2964,20 @@ static void try_eliminate_box_unbox_pair(MVMThreadContext *tc, MVMSpeshGraph *g,
             walk_set_looking_for_unbox(tc, g, bb, ins, unbox_op, decont_op, user);
         user_entry = user_entry->next;
     }
+    if (MVM_spesh_usages_is_used(tc, g, ins->operands[0])) {
+        SeenBox *sb = MVM_malloc(sizeof(SeenBox));
+        sb->bb = bb;
+        sb->ins = ins;
+        MVM_VECTOR_PUSH(pips->seen_box_ins, sb);
```

Comment:
Why use a pointer to a malloc'd structure here? If you'd use spesh_alloc, you wouldn't have to free it, and if you'd use the regular vector form, you'd be able to assign in place.